src/librustc_middle/ty/print/pretty.rs

```diff
@@ -28,9 +28,24 @@ macro_rules! p {
     (@write($($data:expr),+)) => {
         write!(scoped_cx!(), $($data),+)?
     };
+    (@write_full_path($($data:expr),+)) => {
+        with_crate_prefix(|| {
+            write!(scoped_cx!(), $($data),+)
+        })?
+    };
     (@print($x:expr)) => {
         scoped_cx!() = $x.print(scoped_cx!())?
     };
+    (@$method:ident(cond_full_path: $cond:expr, $($arg:expr),*)) => {
+        let res = if $cond {
+            with_crate_prefix(|| {
+                scoped_cx!().$method($($arg),*)
+            })
+        } else {
+            scoped_cx!().$method($($arg),*)
+        };
+        scoped_cx!() = res?
+    };
```

Comment:
I believe you can do these manually around `p!` calls and avoid changing the macro.

Reply:
I tried that approach first, trying to avoid changing this macro, however needing to use `with_crate_prefix(||` closure around `p!` tripped it badly. I don't recall exactly what was the error, but I think it's related to `scoped_cx!()` which makes assumptions that are not workable. I can try again and post about it.

Reply:
Oh I think it's because of the closure nesting, that makes sense. You should be able to remove the users of this post-#74344, as per my other comments, thankfully.